### PR TITLE
(refactor) Replace complex parsing logic from GHEvent.type to GHEvent with static mapping

### PR DESCRIPTION
**Pull request by akashRindhe**
https://github.com/hub4j/github-api/issues/1099

# Description

Replace complex parsing logic from GHEvent.type to GHEvent with static mapping

### CODE REVIEW

File: src/main/java/org/kohsuke/github/GHEvent.java [hunk 1]
```diff
@@ -85,4 +87,50 @@ String symbol() {
...
         return name().toLowerCase(Locale.ENGLISH);
     }
+
+    /**
+     * Representation of GitHub Event Type
+     *
+     * @see <a href="https://docs.github.com/en/developers/webhooks-and-events/github-event-types">GitHub event
+     *      types</a>
+     */
+    public enum GitHubEventType {
```
Comment (bitwiseman) [inline src/main/java/org/kohsuke/github/GHEvent.java:97]: I don't think this needs to be public.  It is only used like an private`HashMap<String, GHEvent>`.  
```suggestion
    enum GitHubEventType {
```

File: src/test/java/org/kohsuke/github/GHEventTest.java [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+        for (GHEvent e : GHEvent.values()) {
+            if (e.name().replace("_", "").equalsIgnoreCase(t)) {
+                return e;
+            }
+        }
+        return GHEvent.UNKNOWN;
+    }
+
+    @Test
+    public void regressionTest() {
```
Comment (bitwiseman) [inline src/test/java/org/kohsuke/github/GHEventTest.java:28]: Add a test for the `GHEvent.UNKNOWN` behavior.

**Review by bitwiseman — CHANGES_REQUESTED** [review]
This is not what I was expecting, but it works, is clearer, and it has to be faster than the old way. 

Minor changes needed.

File: src/main/java/org/kohsuke/github/GHEvent.java [hunk 3]
```diff
@@ -85,4 +87,50 @@ String symbol() {
...
+        IssueCommentEvent(ISSUE_COMMENT),
+        IssuesEvent(ISSUES),
+        MemberEvent(MEMBER),
+        PublicEvent(PUBLIC),
+        PullRequestEvent(PULL_REQUEST),
+        PullRequestReviewEvent(PULL_REQUEST_REVIEW),
+        PullRequestReviewCommentEvent(PULL_REQUEST_REVIEW_COMMENT),
+        PushEvent(PUSH),
+        ReleaseEvent(RELEASE),
+        WatchEvent(WATCH);
```
Comment (bitwiseman) [inline src/main/java/org/kohsuke/github/GHEvent.java:112]:
```suggestion
        WatchEvent(WATCH),
        Unknown(UNKNOWN);
```

File: src/main/java/org/kohsuke/github/GHEvent.java [hunk 4]
```diff
@@ -85,4 +87,50 @@ String symbol() {
...
+         * @param event
+         *            the github event as a string to convert to Event enum
+         * @return GHEvent
+         */
+        public static GHEvent transformToGHEvent(@Nonnull String event) {
+            try {
+                GitHubEventType gitHubEventType = GitHubEventType.valueOf(event);
+                return gitHubEventType.event;
+            } catch (IllegalArgumentException ignored) {
+                return UNKNOWN;
```
Comment (bitwiseman) [inline src/main/java/org/kohsuke/github/GHEvent.java:132]:
```suggestion
            // import static org.kohsuke.github.internal.EnumUtils;
            GitHubEventType gitHubEventType = EnumUtils.getNullableEnumOrDefault(GitHubEventType.class, event, Unknown);                     
            return gitHubEventType.event;
```

**Review by bitwiseman — COMMENTED** [review]
If you use EnumUtils t hat would be better.

**Comment by akashRindhe** [comment]
Thanks for making the changes. I believe everything is resolved now and is good to merge?

**Comment by bitwiseman** [comment]
@akashRindhe 
FYI, I'm sure you could have made the changes yourself, but I felt like getting them done while I was looking at them.   Hope you don't mind.  Thanks for your contributions.

**Comment by akashRindhe** [comment]
> @akashRindhe
> FYI, I'm sure you could have made the changes yourself, but I felt like getting them done while I was looking at them. Hope you don't mind. Thanks for your contributions.

No issues mate. Cheers :)

**Pull request merged**